capabilityreceiver/pkg/capabilityhandler/includedfields.go

```diff
@@ -88,3 +100,177 @@ func checkField(capInt map[string]interface{}, fieldNames []string) bool {
+	for _, extensionPath := range extensionList {
+		extensionName := extensionPath[len(extensionPath)-1]
+		extensionURL := extensionPath[len(extensionPath)-2]
+		extensionObj := endpointmanager.IncludedField{
+			Field:     extensionName,
+			Exists:    checkExtension(capInt, extensionPath, extensionURL),
+			Extension: true,
+		}
+		includedFields = append(includedFields, extensionObj)
+	}
+
+	// Get name of extension with multiple possible locations and create extensionObj with extension name, if the extension exists, and if it is an extension
+	for _, multipleExtensionPath := range multipleFieldsExtensionList {
+		extensionName := multipleExtensionPath[1]
+		extensionURL := multipleExtensionPath[0]
+		extensionObj := endpointmanager.IncludedField{
+			Field:     extensionName,
+			Exists:    checkMultipleFieldsExtension(capInt, extensionURL, extensionName),
+			Extension: true,
+		}
+		includedFields = append(includedFields, extensionObj)
+	}
+
+	return includedFields
+}
+
+// Checks whether the extension is populated in the capability statement given a path of fieldNames
+func checkExtension(capInt map[string]interface{}, fieldNames []string, url string) bool {
+	for index, name := range fieldNames {
+		if capInt[name] == nil {
+			return false
+		}
+
+		field := capInt[name]
+
+		if name == "extension" || name == "modifierExtension" {
+			extensionArr := field.([]interface{})
+			return checkExtensionURL(extensionArr, url)
+		} else if arrContains(arrayFields, name) {
+			fieldArr := field.([]interface{})
+			nextIndex := index + 1
+			length := len(fieldNames)
+			return checkArrFieldExtension(fieldNames[nextIndex:length], fieldArr, url, false)
+		} else {
+			capInt = field.(map[string]interface{})
+		}
+	}
+
+	return false
+}
+
+// Given an array of interface objects, loops through each object to check whether the extension is populated following the path of fieldNames
+func checkArrFieldExtension(fieldNames []string, fieldArr []interface{}, url string, found bool) bool {
+	for _, resource := range fieldArr {
+		name := fieldNames[0]
+		resourceMap := resource.(map[string]interface{})
+		extensionField := resourceMap[name]
+		if extensionField == nil {
+			continue
+		} else if name != "extension" && name != "modifierExtension" && arrContains(arrayFields, name) {
+			fieldArr := extensionField.([]interface{})
+			length := len(fieldNames)
+			found = checkArrFieldExtension(fieldNames[1:length], fieldArr, url, found)
+			if found {
+				return found
+			}
+		} else if name != "extension" && name != "modifierExtension" && !arrContains(arrayFields, name) {
+			extensionField := extensionField.(map[string]interface{})
+			length := len(fieldNames)
+			found = checkExtension(extensionField, fieldNames[1:length], url)
+			if found {
+				return found
+			}
+		} else {
+			extensionArr := extensionField.([]interface{})
+			found = checkExtensionURL(extensionArr, url)
+			if found {
+				return found
+			}
+		}
+	}
+	return found
```

Comment:
I think comments in this function especially would be helpful; it personally took me a while to understand that what was happening was trying to figure out if a field's value was an array or not and calling the correct function accordingly. I think the code itself is clear now that I know what's happening, but a few comments here and there would have helped it click faster, for me at least.